scripts/us_epa/airqualityindex/air_quality_aggregate.py

```diff
@@ -101,12 +107,28 @@ def write_csv(csv_file_path, reader):
 
 
 def request_and_write_csv(csv_file_path, filename):
-    response = requests.get(
-        f'https://aqs.epa.gov/aqsweb/airdata/{filename}.zip')
-    with zipfile.ZipFile(io.BytesIO(response.content)) as zf:
-        with zf.open(f'{filename}.csv', 'r') as infile:
-            reader = csv.DictReader(io.TextIOWrapper(infile, 'utf-8'))
-            write_csv(csv_file_path, reader)
+    try:
+        # Fetch the ZIP archive
+        response = requests.get(
+            f'https://aqs.epa.gov/aqsweb/airdata/{filename}.zip')
+        response.raise_for_status(
+        )  # Raise an exception for non-2xx status codes
+        # Extract the CSV file
+        with zipfile.ZipFile(io.BytesIO(response.content)) as zf:
+            try:
+                with zf.open(f'{filename}.csv', 'r') as infile:
+                    reader = csv.DictReader(io.TextIOWrapper(infile, 'utf-8'))
+                    write_csv(csv_file_path, reader)
+            except FileNotFoundError:
+                raise FileNotFoundError(
+                    f"CSV file '{filename}.csv' not found in the ZIP archive")
+    except (requests.exceptions.RequestException, zipfile.BadZipFile,
+            csv.Error) as e:
+        raise e  # Re-raise the original exception for clearer error handling
+    #Calling method to download zip file locally
+    download_util.download_file_from_url(
+        url=f'https://aqs.epa.gov/aqsweb/airdata/{filename}.zip',
```

Comment:
Can we save the files to input_files subdirectory ?

Reply:
I have modified the code to save the input files under the "input_files" folder